trismart_smartboard_integration/models/smartboard_connector.py

```diff
@@ -2,41 +2,47 @@
 
 class SmartBoard(models.Model):
     _name = "smartboard.connector"
-    _description = "Create SB Project"
+    _description = "SmartBoard Connector"
 
     @api.model
-    def create_project(self, sb_lead_id):
+    def create_project(self, sb_lead_id, project_template_id):
         try:
-            if len(sb_lead_id):
-                partner_exist = self.env['res.partner'].search([('sb_lead_id', '=', sb_lead_id[0])], limit=1)
-                lead_exist = self.env['crm.lead'].search([('sb_lead_id', '=', sb_lead_id[0])], limit=1)
-                project_exist = self.env['project.project'].search([('sb_lead_id', '=', sb_lead_id[0])], limit=1)
+            if sb_lead_id and project_template_id:
+                partner_exist = self.env['res.partner'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                lead_exist = self.env['crm.lead'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                project_exist = self.env['project.project'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                partner_id = []
 
                 if not partner_exist:
                     # Create new partner
-                    partner_id = self.env['res.partner'].create({'name': 'SBLead-' + sb_lead_id[0], 'sb_lead_id': sb_lead_id[0], "sync_status" : '1'})
+                    new_id = self.env['res.partner'].create({'name': 'SBLead-' + sb_lead_id, 'sb_lead_id': sb_lead_id, "sync_status" : '1'})
+                    partner_id.append(new_id)
+
                 if not lead_exist:
                     # Create new lead
-                    lead_id = self.env['crm.lead'].create({'name': 'SBLead-' + sb_lead_id[0], 'sb_lead_id': sb_lead_id[0]})
-
+                    lead_id = self.env['crm.lead'].create({'name': 'SBLead-' + sb_lead_id, 'sb_lead_id': sb_lead_id})
                     # Convert lead to opportunity
-                    customer = self.env['res.partner'].browse(partner_id) if partner_id else partner_exist
-                    for lead in lead_id:
-                        vals = lead._convert_opportunity_data(customer, team_id=False)
-                        lead.write(vals)
+                    if partner_id:
+                        customer = self.env['res.partner'].browse(partner_id[0])
```

Comment:
Do we need partner_id to get the new partner created above since the customer = the new_id?

Reply:
resolved in v3